### PR TITLE
Optimize vcvtps2ph

### PR DESCRIPTION
**Pull request by alyssarosenzweig**
We can avoid a LOT of pointless work with some dedicated IR ops for specifically
overriding the round mode.

Small behaviour change here: we no longer reset FTZ. I think this is a bug fix?
But if it's not it's not hard to fix.

### CODE REVIEW

**Comment by Sonicadvance1** [comment]
> We can avoid a LOT of pointless work with some dedicated IR ops for specifically overriding the round mode.
> 
> Small behaviour change here: we no longer reset FTZ. I think this is a bug fix? But if it's not it's not hard to fix.

Yes, Resetting FTZ (Or any other set FPCR state!) would have been a bug. Good catch.

**Review by Sonicadvance1 — APPROVED** [review]
10/10, would implement dedicated IR ops for things again.

File: FEXCore/Source/Interface/Core/JIT/Arm64/MiscOps.cpp [hunk 1]
```diff
@@ -120,6 +120,37 @@ DEF_OP(SetRoundingMode) {
...
+    return;
+  }
+
+  // Insert the rounding flags, reversing the mode bits as above
+  if (Op->RoundMode == 3) {
+    orr(ARMEmitter::Size::i64Bit, TMP1, Dest, 3 << 22);
+  } else if (Op->RoundMode == 0) {
+    and_(ARMEmitter::Size::i64Bit, TMP1, Dest, ~(3 << 22));
+  } else {
+    and_(ARMEmitter::Size::i64Bit, TMP1, Dest, ~(Op->RoundMode << 22));
```
Comment (lioncash) [inline FEXCore/Source/Interface/Core/JIT/Arm64/MiscOps.cpp:141]: Perhaps have an assert here ensuring the rounding mode is either 1 or 2, so it prevents this from carrying on in the event of invalid values being passed in (either through refactoring in the future, or using this with other things in the future)

Comment (alyssarosenzweig) [reply FEXCore/Source/Interface/Core/JIT/Arm64/MiscOps.cpp:143]: Done. Good idea, thanks.

Comment (Sonicadvance1) [reply FEXCore/Source/Interface/Core/JIT/Arm64/MiscOps.cpp:143]: Could also use the EmitValidation section in IR.json to ensure `RoundMode >= 0 && RoundMode <= 3`

File: FEXCore/Source/Interface/Core/JIT/Arm64/MiscOps.cpp [hunk 2]
```diff
@@ -120,6 +120,39 @@ DEF_OP(SetRoundingMode) {
...
+    return;
+  }
+
+  // Insert the rounding flags, reversing the mode bits as above
+  if (Op->RoundMode == 3) {
+    orr(ARMEmitter::Size::i64Bit, TMP1, Dest, 3 << 22);
+  } else if (Op->RoundMode == 0) {
+    and_(ARMEmitter::Size::i64Bit, TMP1, Dest, ~(3 << 22));
+  } else {
+    LOGMAN_THROW_AA_FMT(Op->RoundMode == 1 || Op->RoundMode == 3, "expect a valid round mode");
```
Comment (lioncash) [inline FEXCore/Source/Interface/Core/JIT/Arm64/MiscOps.cpp:141]:
```suggestion
    LOGMAN_THROW_AA_FMT(Op->RoundMode == 1 || Op->RoundMode == 2, "expect a valid round mode");
```

Comment (alyssarosenzweig) [reply FEXCore/Source/Interface/Core/JIT/Arm64/MiscOps.cpp]: This is awkward... thx

**Review by Sonicadvance1 — APPROVED** [review]

**Pull request merged**